vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,381 @@
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdExternalNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdExternalNetworkCreate,
+		Delete: resourceVcdExternalNetworkDelete,
+		Read:   resourceVcdExternalNetworkRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_scope": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of IP scopes for the network",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"is_inherited": &schema.Schema{
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "True if the IP scope is inherit from parent network",
+						},
+						"gateway": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Gateway of the network",
+							ValidateFunc: singleIP(),
+						},
+						"netmask": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Network mask",
+							ValidateFunc: singleIP(),
+						},
+						"dns1": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Primary DNS server",
+							ValidateFunc: singleIP(),
+						},
+						"dns2": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Secondary DNS server",
+							ValidateFunc: singleIP(),
+						},
+						"dns_suffix": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Dns suffix",
+						},
+						"ip_range": &schema.Schema{
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "IP ranges used for static pool allocation in the network",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"start": &schema.Schema{
+										Type:         schema.TypeString,
+										Required:     true,
+										Description:  "Start address of the IP range",
+										ValidateFunc: singleIP(),
+									},
+									"end": &schema.Schema{
+										Type:         schema.TypeString,
+										Required:     true,
+										Description:  "End address of the IP range",
+										ValidateFunc: singleIP(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"vsphere_networks": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of port groups that back this network. Each referenced DV_PORTGROUP or NETWORK must exist on a vCenter server registered with the system.",
```

Comment:
Is it really only "port groups"? What about "A list of **vSphere networks** or port groups ..."? At least in the GUI it is referred more generically to as "vSphere networks".

Reply:
It depends from what point of view you look. From vCD API they all are port group objects with diff type. From UI they are vsphere networks or similar.

Reply:
OK, port groups they are.